src/main/java/org/jabref/gui/edit/CopyMoreAction.java

```diff
@@ -0,0 +1,228 @@
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.jabref.Globals;
+import org.jabref.JabRefGUI;
+import org.jabref.gui.ClipBoardManager;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefDialogService;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.actions.ActionHelper;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.actions.StandardActions;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.layout.Layout;
+import org.jabref.logic.layout.LayoutHelper;
+import org.jabref.logic.util.OS;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.preferences.JabRefPreferences;
+import org.jabref.preferences.PreferencesService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CopyMoreAction extends SimpleCommand {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CopyMoreAction.class);
+    private StandardActions action;
+    private DialogService dialogService;
+    private StateManager stateManager;
+    private final ClipBoardManager clipBoardManager;
+    private PreferencesService preferencesService;
+
+    public CopyMoreAction(StandardActions action, DialogService dialogService, StateManager stateManager, ClipBoardManager clipBoardManager, PreferencesService preferencesService) {
+        this.action = action;
+        this.dialogService = dialogService;
+        this.stateManager = stateManager;
+        this.clipBoardManager = clipBoardManager;
+        this.preferencesService = preferencesService;
+
+        this.executable.bind(ActionHelper.needsEntriesSelected(stateManager));
+    }
+
+    @Override
+    public void execute() {
+        if (stateManager.getActiveDatabase().isEmpty() || stateManager.getSelectedEntries().isEmpty()) {
+            return;
+        }
+
+        if (!Arrays.asList(
+                StandardActions.COPY_TITLE,
+                StandardActions.COPY_KEY,
+                StandardActions.COPY_CITE_KEY,
+                StandardActions.COPY_KEY_AND_TITLE,
+                StandardActions.COPY_KEY_AND_LINK)
+                  .contains(action)) {
+            return;
+        }
+
+        switch (action) {
+            case COPY_TITLE: copyTitle(); break;
+            case COPY_KEY: copyKey(); break;
+            case COPY_CITE_KEY: copyCiteKey(); break;
+            case COPY_KEY_AND_TITLE: copyKeyAndTitle(); break;
+            case COPY_KEY_AND_LINK: copyKeyAndLink(); break;
+        }
+    }
+
+    private void copyTitle() {
+        List<BibEntry> selectedBibEntries = stateManager.getSelectedEntries();
+
+        List<String> titles = selectedBibEntries.stream()
+                                                .filter(bibEntry -> bibEntry.getTitle().isPresent())
```

Comment:
Maybe a flatmap can be used?

Reply:
I think I do not understand, how? Can you give me a hint?

Reply:
Think, it is OK as is.